File: test/warnings/functions_warning/cpp/FunctionsWarningTest.cpp
#include <string>
#include <vector>

#include "gtest/gtest.h"

#include "test_utils/ZserioErrorOutput.h"

class FunctionsWarningTest : public ::testing::Test
{
protected:
    FunctionsWarningTest()
    :   zserioWarnings("warnings/functions_warning")
    {}

    const test_utils::ZserioErrorOutput zserioWarnings;
};

TEST_F(FunctionsWarningTest, optionalReferencesInFunction)
{
    const std::string warning1 = "optional_references_in_function.zs:11:16: Function "
            "'suspicionFunction' contains reference to optional field 'additionalValue'.";
    ASSERT_TRUE(zserioWarnings.isPresent(warning1));

    const std::string warning2 = "optional_references_in_function.zs:16:16: Function "
            "'autoSuspicionFunction' contains reference to optional field 'autoAdditionalValue'.";
    ASSERT_TRUE(zserioWarnings.isPresent(warning2));
}

File: test/warnings/functions_warning/python/FunctionsWarningTest.py
import unittest

from testutils import getZserioApi, assertWarningsPresent

class FunctionsWarningTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.warnings = {}
        getZserioApi(__file__, "functions_warning.zs", expectedWarnings=2, errorOutputDict=cls.warnings)

    def testOptionalReferencesInFunction(self):
        assertWarningsPresent(self,
            "functions_warning.zs",
            [
                "optional_references_in_function.zs:11:16: Function "
                "'suspicionFunction' contains reference to optional field 'additionalValue'."
            ]
        )

        assertWarningsPresent(self,
            "functions_warning.zs",
            [
                "optional_references_in_function.zs:16:16: Function "
                "'autoSuspicionFunction' contains reference to optional field 'autoAdditionalValue'."
            ]
        )
